modules/executive/components/VoteModal/DefaultView.tsx

```diff
@@ -99,50 +101,56 @@ export default function DefaultVoteModalView({
     const slate = maker.service('web3')._web3.utils.sha3('0x' + encodedParam.slice(-64 * proposals.length));
     const slateAlreadyExists = allSlates && allSlates.findIndex(l => l === slate) > -1;
     const slateOrProposals = slateAlreadyExists ? slate : proposals;
-    const voteTxCreator = voteDelegate
-      ? () => voteDelegate.voteExec(slateOrProposals)
-      : voteProxy
-      ? () => voteProxy.voteExec(slateOrProposals)
-      : () => maker.service('chief').vote(slateOrProposals);
 
-    const txId = await track(voteTxCreator, 'Voting on executive proposal', {
-      pending: txHash => {
-        // if comment included, add to comments db
-        if (comment.length > 0) {
-          const requestBody: ExecutiveCommentsRequestBody = {
-            voterAddress: account?.address || '',
-            delegateAddress: voteDelegate ? voteDelegate.getVoteDelegateAddress() : '',
-            comment: comment,
-            voteProxyAddress: voteProxy ? voteProxy.getProxyAddress() : '',
-            signedMessage: signedMessage,
-            txHash,
-            voterWeight: votingWeight
-          };
-          fetchJson(`/api/comments/executive/add/${proposal.address}?network=${getNetwork()}`, {
-            method: 'POST',
-            body: JSON.stringify(requestBody)
-          })
-            .then(() => {
-              // console.log('comment successfully added');
-              mutateComments();
-            })
-            .catch(() => console.error('failed to add comment'));
-        }
+    const voteCall = Array.isArray(slateOrProposals) ? voteMany : voteOne;
+    const tx = await voteCall(slateOrProposals);
```

Comment:
@adamgoth this is failing with this error: `Error: sending a transaction requires a signer`. I think that we're not updating getContract when we get the new signer after web3-react is connected.